src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -799,6 +804,765 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetExternalIPv6
     return CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_GetThreadNetworkDiagnosticAttributeInfo(
+    chip::AttributeId attributeId, uint8_t ** buffer, uint16_t & ReadLength, chip::TLV::TLVType & type)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    switch (attributeId)
+    {
+    case ZCL_CHANNEL_ATTRIBUTE_ID: {
+        type       = kTLVType_UnsignedInteger;
+        ReadLength = sizeof(uint16_t);
+        *buffer    = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(ReadLength));
+        VerifyOrExit(*buffer != NULL, err = CHIP_ERROR_NO_MEMORY);
+
+        uint16_t channel = static_cast<uint16_t>(otLinkGetChannel(mOTInst));
```

Comment:
Why is this cast safe?

Reply:
The spec specify a uint16 but in thread API channel is a uin8_t. I am juste casting the uin8t value to a uint16 to write it as such in the attribute for spec compliance. Would you rather have me cast it at write time?

Reply:
I would rather the `static_cast` were removed here.  If `otLinkGetChannel` returns `uint8_t` then it can be assigned to a `uint16_t` without any sort of casting.  And if the type `otLinkGetChannel` returns ever changes to something that is not `uint8_t` or `uint16_t`, the compiler would catch that for us at that point, but only if there is no explicit cast here....